integration/compose/src/main/java/com/bumptech/glide/integration/compose/GlideModifier.kt

```diff
@@ -0,0 +1,252 @@
+import androidx.compose.ui.graphics.nativeCanvas
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.layout.Measurable
+import androidx.compose.ui.layout.MeasureResult
+import androidx.compose.ui.layout.MeasureScope
+import androidx.compose.ui.layout.times
+import androidx.compose.ui.node.DrawModifierNode
+import androidx.compose.ui.node.LayoutModifierNode
+import androidx.compose.ui.node.ModifierNodeElement
+import androidx.compose.ui.node.SemanticsModifierNode
+import androidx.compose.ui.node.invalidateDraw
+import androidx.compose.ui.semantics.SemanticsPropertyKey
+import androidx.compose.ui.semantics.SemanticsPropertyReceiver
+import androidx.compose.ui.unit.Constraints
+import androidx.compose.ui.unit.IntOffset
+import androidx.compose.ui.unit.IntSize
+import com.bumptech.glide.RequestBuilder
+import com.bumptech.glide.integration.ktx.AsyncGlideSize
+import com.bumptech.glide.integration.ktx.ExperimentGlideFlows
+import com.bumptech.glide.integration.ktx.ImmediateGlideSize
+import com.bumptech.glide.integration.ktx.InternalGlideApi
+import com.bumptech.glide.integration.ktx.Placeholder
+import com.bumptech.glide.integration.ktx.ResolvableGlideSize
+import com.bumptech.glide.integration.ktx.Resource
+import com.bumptech.glide.integration.ktx.Status
+import com.bumptech.glide.integration.ktx.flowResolvable
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.Job
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.plus
+import kotlin.math.roundToInt
+
+@ExperimentalGlideComposeApi
+@OptIn(InternalGlideApi::class)
+internal fun Modifier.glideNode(
+  requestBuilder: RequestBuilder<Drawable>,
+  size: ResolvableGlideSize,
+  contentScale: ContentScale,
+  alignment: Alignment,
+  colorFilter: ColorFilter?,
+): Modifier =
+  this then GlideNodeElement(requestBuilder, size, contentScale, alignment, colorFilter)
+
+@ExperimentalGlideComposeApi
+@OptIn(InternalGlideApi::class)
+internal data class GlideNodeElement constructor(
+  private val requestBuilder: RequestBuilder<Drawable>,
+  private val size: ResolvableGlideSize,
+  private val contentScale: ContentScale,
+  private val alignment: Alignment,
+  private val colorFilter: ColorFilter?,
+) : ModifierNodeElement<GlideNode>() {
+  override fun create(): GlideNode {
+    val result = GlideNode()
+    result.onNewRequest(requestBuilder, size, contentScale, alignment, colorFilter)
+    return result
+  }
+
+  override fun update(node: GlideNode) {
+    node.onNewRequest(requestBuilder, size, contentScale, alignment, colorFilter)
+  }
+}
+
+@ExperimentalGlideComposeApi
+@OptIn(InternalGlideApi::class)
+internal class GlideNode : DrawModifierNode, LayoutModifierNode, SemanticsModifierNode,
+  Modifier.Node() {
+  private var requestBuilder: RequestBuilder<Drawable>? = null
+  private lateinit var resolvableGlideSize: ResolvableGlideSize
+  private lateinit var contentScale: ContentScale
+  private lateinit var alignment: Alignment
```

Comment:
why make these lateinit? the only place you are constructing this you already have all of these values.

Reply:
This just saves some duplicate code vs:

```
val glideNode = GlideNode(size, scale, alignment)
glideNode.onNewRequest(size, scale, alignment, ...)
```

You could remove the onNewRequest part entirely, but then you're starting a request in a constructor and refactoring that code to be less duplicative. Lesser evil really.